src/app/lib/providers/opensubtitles.js

```diff
@@ -29,11 +29,15 @@
         for (var lang in data) {
             data[lang] = data[lang].url;
         }
-        return Common.sanitize(data);
+        return {subtitle: Common.sanitize(data)};
     };
 
-    OpenSubtitles.prototype.fetch = function (queryParams) {
-        queryParams.extensions = ['srt'];
+    OpenSubtitles.prototype.detail = function (id, attrs) {
```

Comment:
oh, the former implementation just took queryparams (that is id) and passed it to the module, it had a bug trying to extend it, OS is your code i'm not sure what's the best way to handle this. want to take it on from here ?